creator-node/src/apiHelpers.js

```diff
@@ -10,13 +10,23 @@ const { generateTimestampAndSignature } = require('./apiSigning')
 module.exports.handleResponse = (func) => {
   return async function (req, res, next) {
     try {
+      const prometheusRegistry =
+        req.app.get('serviceRegistry').prometheusRegistry
+      const httpRequestTimerMetric = await prometheusRegistry.getMetricInstance(
+        'http_request_duration_seconds'
+      )
+      const endTimer = httpRequestTimerMetric.startTimer()
+
       const resp = await func(req, res, next)
 
       if (!isValidResponse(resp)) {
         throw new Error('Invalid response returned by function')
       }
 
       sendResponse(req, res, resp)
+
+      endTimer({ method: req.method, route: req.path, code: res.statusCode })
```

Comment:
Be mindful that req.path might have a [high cardinality](https://www.notion.so/audiusproject/Metrics-Prometheus-Grafana-ae87cebb12af47a087a7eb2c5fd0040f#bac4d183b1fc465fac4997bc4c078d1d).

We ended up having to strip out some details on [DP's route label](https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/utils/redis_metrics.py#L671-L679).

Reply:
Here are two pieces on Prometheus labels I found looking at their docs. Tossing these here solely since they were related to this thread:

* https://prometheus.io/docs/practices/instrumentation/#use-labels
* https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels